### DISCUSSION TITLE
Is there a way to switch connections within a single SQL notebook between queries?

### DISCUSSION THREAD

**Discussion by Lilit0x** [cmoog/vscode-sql-notebook#83] [Q&A] [answered]
I am creating a end-to-end test with multiple SQL statements within a single SQL notebook and we'd like to switch database connections between certain SQL queries. For example: 

```sql
USE DATABASE myconn_name;  --is something like this allowed?
SELECT * FROM TABLE;
SELECT * FROM TABLE;
SELECT * FROM TABLE;
```

**Comment by cmoog** [comment, marked as answer]
Right now there isn't a way to achieve this. Switching active connections requires invocation in the sidebar.

It sounds like you're trying to use a SQL notebook for an end-to-end testing framework. I don't think it'll be possible for us to make SQL Notebook remain minimal and simple while also serving both the interactive prototyping and automated end-to-end testing use cases. At present, we perform a very minimal parsing of the actual SQL file, only for separating cells. Your proposal would require parsing the SQL into a AST, handling various different dialects, etc. I don't think this is worth the trade off.

**Reply by Lilit0x** [reply]
Thanks for the reply @cmoog , your assessment is correct keeping simplicity is best. Just out of curiosity where are the settings for connection strings stored in VSCode and can I manually edit them?

**Reply by cmoog** [reply]
In an effort to keep configuration files free of secrets (DB connection passwords), I opted for an implementation that keeps the password material inside the VS Code secrets store. The non-secret configuration is stored through the VS Code global state API. I don't think this is directly editable.

**Reply by cmoog** [reply]
In retrospect, this may have been unnecessarily cautious. I'm open to switching to a simple map of `name: connection-string` stored in the human editable settings json file.

**Reply by Lilit0x** [reply]
Yes, we would appreciate an editable version in VSCode settings.

**Reply by Lilit0x** [reply]
Hi @cmoog, could you give us more information whether VSCode settings are editable yet.

**Reply by cmoog** [reply]
They are not editable yet. I'm willing to accept this as an external contribution, or will alternatively start work on it next week.
